### ISSUE TITLE
Then --index option help says it exits after indexing, but doesn't

### ISSUE THREAD

**Issue by madscientist** [MaskRay/ccls#156]
This seems to be a recent change since I didn't see this behavior with a `ccls` I built a week or so ago (last commit was f7c7a3a0bc7a, that worked).  I didn't try to bisect this completely.

In the current HEAD f5604525cbea, adding the `--index` option to the `ccls` command line no longer exits after it's completed indexing as the `--help` output suggests it will.  Instead it just hangs and you have to <kbd>^C</kbd> to exit it.

I ran it under the debugger and broke out when it was hanging and there were 12 threads running, all stuck waiting on `pthread_cond_wait` except for one which is in `pthread_cond_destroy` in the `ccls::MultiQueueWaiter::~MultiQueueWaiter()` dtor.

In case it's useful I've attached the backtraces: [ccls-bt.txt](https://github.com/MaskRay/ccls/files/2666278/ccls-bt.txt).  This is in a release build, unstripped (obviously).  If it's helpful I can do a debug build, or check other things.  Hopefully it's simple enough for anyone to reproduce it and it's not some weird thing on my system.

**Comment by MaskRay** [comment]
This is a data race that is hard to reproduce.

ccls has these types of threads:

* main thread
* "diag"
* "comp"
* "preamble"
* "inderer0" "indexer1" ...

In `-index=/path/to/project` mode, while `Standalone` is running its dtor, one of the three following `ThreadedQueue` shouldn't be destroyed before "preamble" "comp" "diag" exit.

```cpp
struct Semamanager {
...
  ThreadedQueue<std::unique_ptr<CompTask>> comp_tasks;
  ThreadedQueue<DiagTask> diag_tasks;
  ThreadedQueue<PreambleTask> preamble_tasks;
...
```

**Comment by MaskRay** [comment]
This is a longstanding issue I am aware of. cquery/ccls has never implemented a proper "exit" mechanism. So here it is https://github.com/MaskRay/ccls/pull/159

**Comment by MaskRay** [comment]
Fixed by #159
